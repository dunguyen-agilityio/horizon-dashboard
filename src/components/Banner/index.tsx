// Component
import Image from 'next/image';
import { Text } from '@/components';

// Types
import { TEXT_SIZE } from '@/types/text';

const Banner = () => (
  <div className="w-[300px] h-[250px] flex items-end justify-center">
    <div className="w-[228px] h-[190px] bg-gradient-to-br from-purple-750 to-blue-450 rounded-2xl relative">
      <div className="w-[94px] h-[94px] top-[-47px] left-[65px] relative bg-gradient-to-br from-purple-750 to-blue-450 rounded-full border-5 border-white dark:border-indigo">
        <Image
          src="/vector.svg"
          width={41}
          height={41}
          alt="vector"
          className="absolute top-[22px] left-[22px]"
        />
      </div>
      <div className="absolute top-[70px]">
        <Text
          as="h1"
          className="text-medium text-white font-bold leading-9 text-center"
        >
          Upgrade to PRO
        </Text>
        <Text
          as="h2"
          size={TEXT_SIZE.sm}
          className="text-small text-white leading-5 font-medium text-center"
        >
          to get access to all features! Connect with Venus World!
        </Text>
      </div>
    </div>
  </div>
);
export default Banner;
