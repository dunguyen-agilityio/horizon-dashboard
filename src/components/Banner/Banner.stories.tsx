import type { Meta, StoryObj } from '@storybook/react';

// Components
import { Banner } from '@/components';

const meta = {
  title: 'Components/Banner',
  component: Banner,
  tags: ['autodocs'],
} satisfies Meta<typeof Banner>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
