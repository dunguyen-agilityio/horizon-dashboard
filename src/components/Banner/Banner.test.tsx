import { render } from '@testing-library/react';

// Components
import Banner from '.';

describe('Banner component', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Banner />);
    expect(container).toMatchSnapshot();
  });

  it('Should render correct Banner with content', () => {
    const { getByText } = render(<Banner />);

    const titleBanner = getByText('Upgrade to PRO');
    expect(titleBanner).toBeInTheDocument();
  });
});
