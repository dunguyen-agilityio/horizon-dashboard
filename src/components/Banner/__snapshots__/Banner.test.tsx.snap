// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component Should match snapshot 1`] = `
<div>
  <div
    class="w-[300px] h-[250px] flex items-end justify-center"
  >
    <div
      class="w-[228px] h-[190px] bg-gradient-to-br from-purple-750 to-blue-450 rounded-2xl relative"
    >
      <div
        class="w-[94px] h-[94px] top-[-47px] left-[65px] relative bg-gradient-to-br from-purple-750 to-blue-450 rounded-full border-5 border-white dark:border-indigo"
      >
        <img
          alt="vector"
          class="absolute top-[22px] left-[22px]"
          data-nimg="1"
          decoding="async"
          height="41"
          loading="lazy"
          sizes=""
          src="/_next/image?url=%2Fvector.webp&w=96&q=75"
          srcset="/_next/image?url=%2Fvector.webp&w=48&q=75 1x, /_next/image?url=%2Fvector.webp&w=96&q=75 2x"
          style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          width="41"
        />
      </div>
      <div
        class="absolute top-[70px]"
      >
        <h1
          class="text-primary dark:text-white text-medium text-white font-bold leading-9 text-center"
        >
          Upgrade to PRO
        </h1>
        <h2
          class="text-primary dark:text-white text-small text-white leading-5 font-medium text-center"
        >
          to get access to all features! Connect with Venus World!
        </h2>
      </div>
    </div>
  </div>
</div>
`;
